docs/proposals/EdgeSite.md

```diff
@@ -0,0 +1,78 @@
+---
+title: EdgeSite Design
+status: Pending
+authors: 
+  - "@cindyxing"
+approvers:
+  - "@qizha"
+  - "@kevin-wangzefeng"
+  - "@m1093782566"
+---
+# EdgeSite: Lightweight Cluster at edge
+
+## Abstract
+In Edge computing, there are scenarios where customers would like to have an whole cluster installed at edge location. As a result, 
+admins/users can leverage the local control plane to implement management functionalities and take advantages of all edge computing's benefits. 
+
+This design doc is to enable customers deploy and run lightweight clusters at edge. 
+
+## Motivation
+For these use cases, when considering large set of edge locations, the resource usage of management plane adding together will be high. 
+K3s is an opensource project enabling lightweight kubernetes cluster including the light control plane and worker components. 
+KubeEdge is the other opensource project enabling lightweight k8s client on the edge nodes. The KubeEdge agent runtime memory footprint is 
+10MB and it supports kubernetes types. Meanwhile KubEdge is targeted for Edge/IOT computing with many functionality enablements. 
```

Comment:
> Sorry for joining the review late.
> The edge-site cases is quite important scenario in edge computing, big +1 to the overall idea.
> 
> For the design detail, I have 2 major questions here:
> 
> 1. It's not clear in the architecture that what components are from k3s, what can be conformant Kubernetes components. One of the overall external dependencies so far of KubeEdge is just a conformant Kubernetes, it's better we keep this and not tied to a specific implementation when dealing with edge-site cases.
> 2. As for the location of edgecontroller and removal of cloudhub / edgehub, why not keep edgecontroller in the control plane at edge and keep cloudhub / edgehub to deal with behind-firewall discovery / communication issue? I would like to see pros and cons listed in comparison between these two solutions.

Yes, in principal Edgesite doesn't couple with K3s. In fact, the Edgesite does conform to the K8s API that it can work with a normal K8s control plane. Let me try to update the architecture a bit and make it clearer. Thanks for pointing this out. 

For the 2nd comment, not sure what you meant. The controller is at edge side. For edgesite scenario, it is to deploy a whole cluster at edge. Why would there be a need of  firewall? Maybe the code PR can explain. I'll update the architecture diagram to make it more obvious.